internal/pkg/manifest/validate.go

```diff
@@ -1473,6 +1474,71 @@ func (q SQSQueue) validate() error {
 	return nil
 }
 
+// validate returns nil if FIFOAdvanceConfig is configured correctly.
+func (q FIFOAdvanceConfig) validate() error {
+	if q.IsEmpty() {
+		return nil
+	}
+
+	if err := q.validateHighThroughputFIFO(); err != nil {
+		return err
+	}
+	if err := q.validateDeduplicationScope(); err != nil {
+		return err
+	}
+	if err := q.validateFIFOThroughputLimit(); err != nil {
+		return err
+	}
+	if aws.StringValue(q.FIFOThroughputLimit) == sqsFIFOThroughputLimitPerMessageGroupID && aws.StringValue(q.DeduplicationScope) == sqsDeduplicationScopeQueue {
+		return fmt.Errorf(`"fifo_throughput_limit" must be set to "perQueue" when "deduplication_scope" is set to "queue"`)
+	}
+	return nil
+}
+
+// validateFIFO returns nil if FIFOAdvanceConfigOrBool is configured correctly.
+func (q FIFOAdvanceConfigOrBool) validate() error {
+	if q.Enable != nil {
+		return nil
+	}
+	return q.Advanced.validate()
```

Comment:
```suggestion
	if q.isEmpty() {
		return nil
	}
	return q.Advanced.validate()
```

nit: imo I feel like we don't want to skip `q.Advanced.validate()` if `q.Enable != nil`.. so far my understanding for `validate` is that it should just validate every fields without the assumption that some fields being non-empty implies that the other must be empty